pkg/util/util.go

```diff
@@ -41,9 +41,29 @@ func ToCanonicalName(name string) string {
 	return strings.Join(nameParts, ".")
 }
 
+// OpenAPICanonicalTypeNamer is an interface for models without Go types to seed model name.
+//
+// OpenAPI canonical names are REST resource friendly names, converted from full package path
+// for Go types or from group/version/kind (GVK) for custom resource definition (CRD)
+// Examples:
+//	Go type:                k8s.io/api/core/v1.Pod
+//	OpenAPI canonical name: io.k8s.api.core.v1.Pod
```

Comment:
I don't see the reverse dns transformation happening below... just the package extraction and /vendor/ stripping

Reply:
you're right... I'm mixing things up here

the namer is just for providing `package.type` name

Reply:
updated. 

I will also switch 
https://github.com/kubernetes/kubernetes/blob/91d2bd3951f3a2e7290ef0cf5ec546ba228944d7/staging/src/k8s.io/apiserver/pkg/endpoints/openapi/openapi.go#L124
to use `ToRESTFriendlyName` here